R/create_protocol.R

```diff
@@ -54,36 +84,49 @@ create_sfp <- function(
   assert_that(is.string(revision))
   theme <- match.arg(theme)
   language <- match.arg(language)
+  if (!missing(from_docx)) {
+    assert_that(is.string(from_docx),
+                file.exists(from_docx))
+  }
+  protocol_type <- "sfp"
+  if (!missing(protocol_number)) {
+    assert_that(
+      is.string(protocol_number),
+      !(protocol_number %in% get_protocolnumbers(protocol_type = protocol_type))
+      )
+  }
 
   # create protocol name
+  if (missing(protocol_number)) {
+    protocol_leading_number <- themes_df[themes_df$theme == theme,
+                                         "theme_number"]
 
-  protocol_type <- "sfp"
+    all_numbers <- get_protocolnumbers(protocol_type = protocol_type)
 
-  protocol_leading_number <- themes_df[themes_df$theme == theme,
-                                       "theme_number"]
+    if (length(all_numbers) == 0) {
+      protocol_trailing_number <- "01"
+    } else {
+      numbers <- str_subset(all_numbers, paste0("^", protocol_leading_number))
+      protocol_trailing_number <- max(
+        as.integer(
+          str_extract(numbers, "\\d{2}$"))
+      ) + 1
+      protocol_trailing_number <- formatC(protocol_trailing_number,
+                                          width = 2, format = "d", flag = "0")
+    }
 
-  all_numbers <- get_protocolnumbers(protocol_type = protocol_type)
-
-  if (length(all_numbers) == 0) {
-    protocol_trailing_number <- "01"
-  } else {
-    numbers <- str_subset(all_numbers, paste0("^", protocol_leading_number))
-    protocol_trailing_number <- max(
-      as.integer(
-        str_extract(numbers, "\\d{2}$"))
-    ) + 1
-    protocol_trailing_number <- formatC(protocol_trailing_number,
-                                   width = 2, format = "d", flag = "0")
+    protocol_number <- paste0(protocol_leading_number,
+                              protocol_trailing_number)
   }
 
-  protocol_number <- paste0(protocol_leading_number,
-                            protocol_trailing_number)
-
   short_title <- tolower(short_title)
   short_title <- str_replace_all(short_title, " ", "-")
   short_titles <- get_short_titles(protocol_type = protocol_type)
   assert_that(!(short_title %in% short_titles),
-              msg = "The given short title already exists. Give a short title that is not in use. Use get_short_titles() to get an overview of short titles that are in use.")
+              msg = "The given short title already exists.
+              Give a short title that is not in use.
+              Use get_short_titles() to get an overview of short titles
+              that are in use.")
 
   protocol_code <- paste0(protocol_type, "-", protocol_number)
   folder_name <- paste0(protocol_code, "_", short_title)
```

Comment:
Not sure if we should add language to the folder name. If the exact same protocol is provided in Dutch and English, we should consider maintaining them in the same place. Don't know how this would be done technically. Maybe the regular text in each of the Rmarkdown could be placed in chunks and have chunk options `eval = language == "nl", results = "asis"` or something.

Reply:
Maybe you can split directory in `src` and not in `doc` if that's easier (but that may be strange). I would rather not combine more than 1 language in the same Rmd document, if that's what you suggest (especially for longer texts). But perhaps it's a matter of taste, I think there are pros and cons. If languages are separated between Rmd files, then I do suggest two keep them in separate directories to keep it simple - those directories would still have the same protocol number (right?) and hence be sorted conveniently.

Reply:
OK, I follow your suggestion, but I think I will have to make some changes because that would mean that the same protocol number is allowed in multiple places. There is currently a check that asserts if the protocol number exists already.

Reply:
I'm confident that you will make the best choice regardless of what I say (you have better overview); I'm just there to challenge your mind :wink:

Reply:
See 
3c2d588